HIF.md

```diff
@@ -0,0 +1,739 @@
+ - `uint8_t dwell_interval` (from US-IE)
+
+Only present if `FHSS_TYPE_LFN_UC`:
+
+ - `uint64_t lutt_timestamp_us`  
+    Timestamp associated with the last received LUTT-IE from this node. The host
+    will use the `timestamp_us` value from [`CNF_DATA_TX`][tx-cnf] and
+    [`IND_DATA_RX`][rx].
+ - `uint16_t slot` (from LUTT-IE)
+ - `uint24_t interval_offset_ms` (from LUTT-IE)
+ - `uint24_t interval_ms` (from LUS-IE)
+
+Only present if `FHSS_TYPE_LFN_PA`:
+
+ - `uint64_t lnd_timestamp_us`  
+    Timestamp associated with the last received LND-IE from this node. The host
+    will use the `timestamp_us` value from [`IND_DATA_RX`][rx].
+ - `uint24_t response_delay_ms` (from LND-IE)
+ - `uint8_t  slot_duration_ms` (from LND-IE)
+ - `uint8_t  slot_count` (from LND-IE)
+ - `uint16_t slot_first` (from LND-IE)
+
+Only present if `FHSS_DEFAULT == 0`:
+
+ - `struct chan_seq`  
+   [Channel sequence][chan-seq] description.
+
+For each bit set in `FRAME_COUNTERS`:
+
+ - `uint32_t frame_counter`  
+   Minimum frame counter to accept when verifying acknowlegment frames. The
+   associated key index maps to the bit offset (from 1 to 7). See
+   ["Security"][sec].
+
+Only present if `MODE_SWITCH`:
+
+ - `struct rate_config[4]`  
+    Fixed length and ordered array of rates to attempt until the transmission is
+    succesful, or all rates been tried.
+     - `uint8_t phy_mode_id`: Wi-SUN PhyModeId, must be in the group selected
+        with [`SET_RADIO`][rf-set].
+     - `uint8_t tx_attempts`: Maximum number of attempts for this rate, try the
+        next entry once this is exceeded.
+     - `int8_t tx_power_dbm`: TX power to use with this entry, saturates to the
+        value configured with [`SET_RADIO_TX_POWER`][rf-pow].
+
+### `0x12 CNF_DATA_TX`
+
+Status of an earlier data request ([`REQ_DATA_TX`][tx-req]). Also returns the
+acknowledge frame if relevant.
+
+ - `uint8_t handle`  
+    Frame handle, as provided in [`REQ_DATA_TX`][tx-req].
+
+ - `uint8_t status`  
+    See the table below for a list of status codes. The fields below may be
+    invalid if `status != 0`.
+
+ - `uint16_t frame_len`  
+    Length of the IEEE 802.15.4 acknowlegment frame if received.
+
+ - `uint8_t frame[]`  
+    Unencrypted IEEE 802.15.4 acknowlegment frame (see ["Security"][sec]).
+    <!-- TODO: Explain EDFE. -->
+
+ - `uint64_t timestamp_us`  
+    The timestamp (relative to the date of the RCP reset) when the frame have
+    been received.
+
+ - `uint8_t lqi`  
+    Received Link Quality Indicator (LQI) as reported by [RAIL][lqi] (ACK
+    received only).
+
+ - `int8_t rx_power_dbm`  
+    Received power in dBm as reported by [RAIL][rssi] (ACK received only).
+
+ - `uint32_t frame_counter`  
+    Frame counter used in the successful transmission (see `frame_counter` in
+    [`SEC_SET_KEY`][key]). Only valid if the request included an auxiliary
+    security header.
+    <!-- TODO: Return the last used frame counter, even if unsuccessful. -->
```

Comment:
This is more a TODO in RCP code. The issue currently is that the frame counter is incremented for all retries, but it is not returned to the host in case of TX failure.